clients/drcachesim/common/zipfile_istream.h

```diff
@@ -103,25 +104,42 @@ class zipfile_istreambuf_t : public std::basic_streambuf<char, std::char_traits<
         }
         return gptr() - eback();
     }
+    std::string
+    open_component(const std::string &name)
+    {
+        if (unzCloseCurrentFile(file_) != UNZ_OK)
+            return "Failed to close current component";
```

Comment:
Wasn't there some error struct that gave details for why an unz operation failed? Should we print that?

Reply:
The return value has some different error codes defined.  I added that to the returned string here and in the two cases below.